cli/tools/upgrade.rs

```diff
@@ -125,11 +125,19 @@ pub fn check_for_upgrades(cache_dir: PathBuf) {
 
 pub async fn upgrade(upgrade_flags: UpgradeFlags) -> Result<(), AnyError> {
   let old_exe_path = std::env::current_exe()?;
-  let permissions = fs::metadata(&old_exe_path)?.permissions();
+  #[allow(unused_variables)]
+  let metadata = fs::metadata(&old_exe_path)?;
+  let permissions = metadata.permissions();
 
   if permissions.readonly() {
     bail!("You do not have write permission to {:?}", old_exe_path);
```

Comment:
We shouldn't be showing a debug output here (I forget what that looks like, but let's go with .display())

```suggestion
    bail!("You do not have write permission to {}", old_exe_path.display());
```